src/vcpkg/vcpkgpaths.cpp

```diff
@@ -1220,53 +1162,94 @@ namespace vcpkg
+
+    ExpectedL<Unit> VcpkgPaths::git_read_tree(const Path& destination, StringView tree, const Path& dot_git_dir) const
     {
         auto& fs = get_filesystem();
-        fs.create_directories(m_pimpl->m_registries_git_trees, VCPKG_LINE_INFO);
-
-        auto git_tree_final = m_pimpl->m_registries_git_trees / object;
-        if (fs.exists(git_tree_final, IgnoreErrors{}))
+        std::error_code ec;
+        auto pid = get_process_id();
+        Path git_tree_temp = fmt::format("{}_{}.tmp", destination, pid);
+        Path git_tree_index = fmt::format("{}_{}.index", destination, pid);
+        auto parent = destination.parent_path();
+        if (!parent.empty())
         {
-            return git_tree_final;
+            fs.create_directories(parent, ec);
+            if (ec)
+            {
+                return format_filesystem_call_error(ec, "create_directories", {parent});
+            }
         }
 
-        auto pid = get_process_id();
+        fs.remove_all(git_tree_temp, ec);
+        if (ec)
+        {
+            return format_filesystem_call_error(ec, "remove_all", {git_tree_temp});
+        }
 
-        Path git_tree_temp = fmt::format("{}.tmp{}", git_tree_final, pid);
-        Path git_tree_temp_tar = git_tree_temp.native() + ".tar";
-        fs.remove_all(git_tree_temp, VCPKG_LINE_INFO);
-        fs.create_directory(git_tree_temp, VCPKG_LINE_INFO);
+        fs.create_directory(git_tree_temp, ec);
+        if (ec)
+        {
+            return format_filesystem_call_error(ec, "create_directory", {git_tree_temp});
+        }
 
-        const auto& dot_git_dir = m_pimpl->m_registries_dot_git_dir;
-        Command git_archive = git_cmd_builder(dot_git_dir, m_pimpl->m_registries_work_tree_dir)
-                                  .string_arg("archive")
-                                  .string_arg("--format")
-                                  .string_arg("tar")
-                                  .string_arg(object)
-                                  .string_arg("--output")
-                                  .string_arg(git_tree_temp_tar);
-        auto maybe_git_archive_output = flatten(cmd_execute_and_capture_output(git_archive), Tools::GIT);
-        if (!maybe_git_archive_output)
-        {
-            return msg::format_error(msgGitCommandFailed, msg::command_line = git_archive.command_line())
-                .append_raw('\n')
-                .append(std::move(maybe_git_archive_output).error());
+        fs.remove_all(destination, ec);
+        if (ec)
+        {
+            return format_filesystem_call_error(ec, "remove_all", {destination});
         }
 
-        extract_tar_cmake(get_tool_exe(Tools::CMAKE, stdout_sink), git_tree_temp_tar, git_tree_temp);
-        // Attempt to remove temporary files, though non-critical.
-        fs.remove(git_tree_temp_tar, IgnoreErrors{});
+        Command git_archive = git_cmd_builder(dot_git_dir, git_tree_temp)
```

Comment:
Should this be refactored into `base/git.cpp`?

Reply:
I actually did a bunch of that and backed it out because it was more changes that way. I would like to see an overall bulk move of the git stuff into there but that change seems independent of this one...